### ISSUE TITLE
[Bug] TAS Recorder crashes Everest when using a different codec than H.264 and AAC

### ISSUE THREAD

**Issue by fsvgm777** [psyGamer/TASRecorder#5]
See title.

What happens is that instead of the game running (with the Recorder popping up), it freezes before eventually closing itself.

Attached is a log file showing that it spams lines stating

`(10/14/2023 15:32:52) [Everest] [Error] [TASRecorder] FFmpeg Error - Failed encoding frame: Invalid argument [-22]`. It is not impossible that the mod tries to use the `-preset` argument regardless if H.264 is selected or not. I used MKV, with the VP9 video and the AAC audio codec.
[log_20231014_153252.txt](https://github.com/psyGamer/TASRecorder/files/12906197/log_20231014_153252.txt)

**Comment by psyGamer** [comment]
Since you're on Windows, it could be that the FFmpeg libraries which are shipped with TASRecorder don't support other codecs. Ill have to test if they work on my VM.

**Comment by fsvgm777** [comment]
I can check in my Linux VM real quick to see if that's the culprit.

**Comment by psyGamer** [comment]
I develop it on Linux and it works there.
Just tested it in the VM and it didn't work so I assume the libraries don't support them.
I'll see if I can find "better" libraries.

**Comment by psyGamer** [comment]
@fsvgm777 I updated the libraries and they seem to work on my VM.
Can you confirm that this build works?
[TASRecorder.zip](https://github.com/psyGamer/TASRecorder/files/12906847/TASRecorder.zip)

**Comment by fsvgm777** [comment]
Alas, these still don't work on Windows.

I also tried getting TAS Recorder to work on Linux, but without any success (I cannot find the output anywhere, and the terminal spits out the following when I do start_recording):

```
System.EntryPointNotFoundException: Unable to find an entry point named 'av_channel_layout_copy' in shared library 'avutil'.
   at FFmpeg.FFmpeg.av_channel_layout_copy(AVChannelLayout* dst, AVChannelLayout* src)
   at Celeste.Mod.TASRecorder.Encoder.AddStream(AVCodec* codec, OutputStream& outStream, AVCodecID codecID)
   at Celeste.Mod.TASRecorder.Encoder..ctor(String fileName)
   at Celeste.Mod.TASRecorder.TASRecorderModule.StartRecording(Int32 frames, String fileName)
   at Celeste.Mod.TASRecorder.TASRecorderModule.CmdStartRecording(Int32 frames)
TypeLoadException.TypeName:
```

For the record, this is on a Kubuntu 22.04 VM, which ships an outdated version of ffmpeg in its package repo.

**Comment by psyGamer** [comment]
Hmmm, but it worked on my VM...

As for getting it to run on Linux, if they ship an outdated FFmpeg version, that's not my problem.
Tho there should be libraries installed if it can't find system libraries, so you'd need to uninstall those.

**Comment by fsvgm777** [comment]
Right, it doesn't actually rely on FFmpeg itself, but its libraries. Unfortunately, uninstalling the already installed libraries in my Linux VM is not really an option for me, as it'd also uninstall the graphical file manager as well as VLC, etc...

**Comment by psyGamer** [comment]
This build removes the system library attempt.
Note that it might not actually work since i just copied my library files, which could depend on stuff which isnt actually installed..
[TASRecorder.zip](https://github.com/psyGamer/TASRecorder/files/12907906/TASRecorder.zip)

**Comment by fsvgm777** [comment]
Tested on Windows, and it works fine with other codecs (VP9+Opus, for instance)....albeit slowly, but that's to be expected (at least for that combination).

**Comment by psyGamer** [comment]
Fixed in 6d840f640da97871800eb24a480e3447b0e28d9d
